perllib/FixMyStreet/App/Controller/Admin/AreaStats.pm

```diff
@@ -0,0 +1,208 @@
+
+    my $user = $c->user;
+
+    if ($user->is_superuser) {
+        return;
+    }
+
+    if ($body_id and $user->from_body->id eq $body_id) {
+        if (not $user->area_id) {
+            return;
+        } elsif ($area_id and $user->area_id eq $area_id) {
+            return;
+        }
+    }
+
+    $c->detach( '/page_error_404_not_found' );
+}
+
+sub body_base : Chained('/') : PathPart('admin/areastats') : CaptureArgs(1) {
+    my ($self, $c, $body_id) = @_;
+    $c->stash->{body_id} = $body_id;
+    $c->forward('/admin/lookup_body');
+    $c->stash->{areas} = mySociety::MaPit::call('area/children', [ $body_id ] );
+}
+
+sub body_stats : Chained('body_base') : PathPart('') : Args(0) {
+    my ($self, $c) = @_;
+    my $user = $c->user;
+
+    $c->forward('check_user', [$c->stash->{body_id}]);
+    if ($c->get_param('area')) {
+        $c->stash->{area_id} = $c->get_param('area');
+    } else {
+        $c->stash->{area_id} = $c->stash->{body}->id;
+    }
+    $c->forward('stats');
+}
+
+sub ward_stats : Chained('body_base') :PathPart('') : Args(1) {
+    my ($self, $c, $area_id) = @_;
+    my $user = $c->user;
+
+    $c->forward('check_user', [$c->stash->{body_id}, $area_id]);
+
+    $c->stash->{area_id} = $area_id;
+    $c->forward('stats');
+}
+
+sub stats : Private {
+    my ($self, $c) = @_;
+
+    my $date = DateTime->now->subtract(days => 30);
+    my $area_id = $c->stash->{area_id};
+    my $area = mySociety::MaPit::call('area', $area_id );
+    my $user = $c->user;
+
+    $c->forward('/admin/fetch_contacts');
+
+    $c->stash->{template} = 'admin/areastats/area.html';
+    if ($area->{name}) {
+        $c->stash->{area} = $area;
+
+        my $dtf = $c->model('DB')->storage->datetime_parser;
+        my $time = $dtf->format_datetime($date);
+
+        my $params = {
+            'problem.areas' => { like => "%,$area_id,%" },
+            'me.confirmed' => { '>=', $time },
+        };
+
+        my $comments = $c->model('DB::Comment')->search(
+            $params,
+            {
+                group_by => [ 'problem_state' ],
+                select   => [ 'problem_state', { count => 'me.id' } ],
+                as       => [ qw/state state_count/ ],
+                join     => 'problem'
+            }
+        );
+
+        $params = {
+            areas => { like => "%,$area_id,%" },
+        };
+
+        my $problems = $c->model('DB::Problem')->search(
+            $params,
+            {
+                group_by => [ 'category', 'state' ],
+                select   => [ 'category', 'state', { count => 'me.id' } ],
+                as       => [ qw/category state state_count/ ],
+            }
+        );
+
+        my %by_category = map { $_->category => {} } $c->stash->{contacts}->all;
+        my %recent_by_category = map { $_->category => 0 } $c->stash->{contacts}->all;
+
+        my $state_map = {};
+
+        $state_map->{$_} = 'open' foreach FixMyStreet::DB::Result::Problem->open_states;
```

Comment:
This will include planned/action scheduled which are also under scheduled, is that as expected?